### PR TITLE
chore: deprecated useSimpleBuildTest, waitForQuiescence and calendarAccessAuthorized

### PR DESCRIPTION
**Pull request by KazuCocoa**
Instead of https://github.com/appium/WebDriverAgent/pull/887

```
[AppiumDriver@6da2] Appium v2.5.1 creating new XCUITestDriver (v7.11.2) session
[AppiumDriver@6da2] Checking BaseDriver versions for Appium and XCUITestDriver
[AppiumDriver@6da2] Appium's BaseDriver version is 9.5.2
[AppiumDriver@6da2] XCUITestDriver's BaseDriver version is 9.5.2
[XCUITestDriver@31a2] Creating session with W3C capabilities: {
[XCUITestDriver@31a2]   "alwaysMatch": {
[XCUITestDriver@31a2]     "platformName": "ios",
[XCUITestDriver@31a2]     "appium:automationName": "xcuitest",
[XCUITestDriver@31a2]     "appium:platformVersion": "17.2",
[XCUITestDriver@31a2]     "appium:deviceName": "iPhone 15 Pro",
[XCUITestDriver@31a2]     "appium:udid": "242bfcc998fd156df0bd0ad1dde8ab8e0a032114",
[XCUITestDriver@31a2]     "appium:useSimpleBuildTest": true,
[XCUITestDriver@31a2]     "appium:webDriverAgentUrl": "http://192.168.4.30:8100"
[XCUITestDriver@31a2]   },
[XCUITestDriver@31a2]   "firstMatch": [
[XCUITestDriver@31a2]     {}
[XCUITestDriver@31a2]   ]
[XCUITestDriver@31a2] }
[XCUITestDriver@31a2] The desired capabilities include neither an app nor a bundleId. WebDriverAgent will be started without the default app
[XCUITestDriver@31a2 (647fdf44)] Session created with session id: 647fdf44-2b31-45f5-b989-e2701ea34851
[XCUITestDriver@31a2 (647fdf44)] 'appium:useSimpleBuildTest' is deprecated, which was to build the WebDriverAgent project with 'build' and 'test' legacy commands for older than Xcode 8. Please remove the capability to use newer 'build-for-testing' and 'test-without-building'.
[XCUITest] Current user: 'kazu'
[XCUITestDriver@31a2 (647fdf44)] Skipping checking of the real devices availability since the session specifies appium:webDriverAgentUrl
[XCUITestDriver@31a2 (647fdf44)] Creating iDevice object with udid '242bfcc998fd156df0bd0ad1dde8ab8e0a032114'
```

### CODE REVIEW

File: lib/driver.js [hunk 1]
```diff
@@ -416,6 +426,8 @@ export class XCUITestDriver extends BaseDriver {
         this.validateDesiredCaps({...caps, ...this.cliArgs});
       }
 
+      this.printDeprecatedCaps();
```
Comment (mykola-mokhnach) [inline lib/driver.js:429]: it also makes sense to mark the capability definition in desired-caps as `@deprecated`

Comment (KazuCocoa) [reply lib/driver.js:428]: with `useSimpleBuildTest`

```
[XCUITestDriver@7d57]   ]
[XCUITestDriver@7d57] }
[BaseDriver] useSimpleBuildTest is a deprecated capability
[BaseDriver] useSimpleBuildTest is a deprecated capability
[XCUITestDriver@7d57] The desired capabilities include neither an app nor a bundleId. WebDriverAgent will be started without the default app
[XCUITestDriver@7d57 (0d9a20c1)] Session created with session id: 0d9a20c1-6379-4a0e-8033-2238f4bba6d2
```


without `useSimpleBuildTest`

```
[XCUITestDriver@8e46] }
[BaseDriver] useSimpleBuildTest is a deprecated capability
[BaseDriver] useSimpleBuildTest is a deprecated capability
[BaseDriver] useSimpleBuildTest is a deprecated capability
[XCUITestDriver@8e46] The desired capabilities include neither an app nor a bundleId. WebDriverAgent will be started without the default app
[XCUITestDriver@8e46 (555c4c93)] Session created with session id: 555c4c93-b2d3-47b1-aaf4-a24783cb9ccc
```

interesting.. maybe some behavior in the base driver has wrong handling

Comment (mykola-mokhnach) [reply lib/driver.js:428]: yes, it might be a bug...
Could you please check it when you have time?

File: lib/driver.js [hunk 2]
```diff
@@ -406,6 +406,15 @@ export class XCUITestDriver extends BaseDriver {
     return !(CAP_NAMES_NO_XCODEBUILD_REQUIRED.some((x) => Boolean(this.opts[x])));
   }
 
+  /**
+   * Print deprecated capabilities.
+   */
+  printDeprecatedCaps() {
+    if (!_.isUndefined(this.opts.useSimpleBuildTest)) {
+      this.log.warn("The capability 'appium:useSimpleBuildTest' is deprecated and will be removed.");
```
Comment (mykola-mokhnach) [inline lib/driver.js:414]: do we still need this message if `deprecated` above is set true?

Comment (mykola-mokhnach) [reply lib/driver.js:414]: perhaps it would be just enough to rely on the base driver behaviour. I assume it should be aligned with other drivers

Comment (KazuCocoa) [reply lib/driver.js:414]: initially though to add more context why, but yes, could be removed. Actually older Appium may show deprecation message by basedriver for non-usage thus I thought this also helped. But I can remove them.

File: docs/reference/capabilities.md [hunk 3]
```diff
@@ -78,7 +78,7 @@ about capabilities, refer to the [Appium documentation](https://appium.io/docs/e
...
 |`appium:simpleIsVisibleCheck`|Use native methods for determining visibility of elements. In some cases this takes a long time. Setting this capability to `false` will cause the system to use the position and size of elements to make sure they are visible on the screen. This can, however, lead to false results in some situations. Defaults to `false`. | `true`, `false`|
-|`appium:waitForQuiescence`| It allows to turn on/off waiting for application quiescence in `WebDriverAgent`, while performing queries. The default value is `true`. You can avoid [this kind of issues](https://github.com/appium/appium/issues/11132) if you turn it off. Consider using `waitForIdleTimeout` capability instead for this purpose since Appium 1.20.0 | `false` |
+|`appium:waitForQuiescence`| **Deprecated** It allows to turn on/off waiting for application quiescence in `WebDriverAgent`, while performing queries. The default value is `true`. You can avoid [this kind of issues](https://github.com/appium/appium/issues/11132) if you turn it off. Consider using `waitForIdleTimeout` capability instead for this purpose since Appium 1.20.0 | `false` |
```
Comment (KazuCocoa) [inline docs/reference/capabilities.md:81]: It looks like we expect to use `waitForIdleTimeout` instead thus I think we can make this caps deprecate explicitly

File: docs/reference/capabilities.md [hunk 4]
```diff
@@ -98,7 +98,7 @@ about capabilities, refer to the [Appium documentation](https://appium.io/docs/e
...
 |`appium:skipSyncUiDialogTranslation`|Set this option to `true` in order to skip synchronizing UI dialogs translation. While this option might leave some system UI alerts untranslated, it helps to avoid unexpected side effects (see [this issue](https://github.com/appium/appium/issues/19440) for more details). It is set to `false` by default. |`true` or `false`|
-| **Deprecated** `appium:calendarAccessAuthorized`|This capability is obsolete. Please use `appium:permissions` one instead with the `calendar` key.|`true` or `false`|
+| `appium:calendarAccessAuthorized`|  **Deprecated** This capability is obsolete. Please use `appium:permissions` one instead with the `calendar` key.|`true` or `false`|
```
Comment (KazuCocoa) [inline docs/reference/capabilities.md:101]: Just move the `Deprecated` to the description. Which is better...?

Comment (KazuCocoa) [reply docs/reference/capabilities.md:101]: hm, maybe caps name itself might be better. Let me follow it

File: lib/desired-caps.js [hunk 5]
```diff
@@ -202,12 +202,15 @@ const desiredCapConstraints = /** @type {const} */ ({
   },
   calendarAccessAuthorized: {
     isBoolean: true,
+    deprecated: true
```
Comment (KazuCocoa) [inline lib/desired-caps.js:205]: we already addressed this is deprecated in the capabilities.md

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**